network.go

```diff
@@ -574,9 +580,91 @@ func (s *sandbox) updateRoute(netHandle *netlink.Handle, route *types.Route, add
 /////////
 // DNS //
 /////////
+func mountFile(source, destination, fsType string, flags int, options string) error {
+	agentLog.WithFields(logrus.Fields{
+		"mount-source":      source,
+		"mount-destination": destination,
+		"mount-fstype":      fsType,
+		"mount-flags":       flags,
+		"mount-options":     options,
+	}).Debug()
+
+	if source == "" {
+		return fmt.Errorf("need mount source")
+	}
+
+	if destination == "" {
+		return fmt.Errorf("need mount destination")
+	}
+
+	if fsType == "" {
+		return fmt.Errorf("need mount FS type")
+	}
+	var err error
+
+	_, err = os.Stat(destination)
+	if os.IsNotExist(err) {
+		os.Create(destination)
+	}
+
+	_, err = os.Stat(source)
+	if os.IsNotExist(err) {
+		return grpcStatus.Errorf(codes.Internal, "Mount source %v does not exist. Error: %v",
+			source, err)
+	}
+	if err = syscall.Mount(source, destination,
+		fsType, uintptr(flags), options); err != nil {
+		return grpcStatus.Errorf(codes.Internal, "Could not mount %v to %v: %v",
+			source, destination, err)
+	}
 
-func setupDNS(dns []string) error {
 	return nil
+
+}
+
+func traverseFiles(files *[]string) filepath.WalkFunc {
+	return func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		*files = append(*files, path)
+		return nil
+	}
+}
+
+func getSandboxDNSFile(sandboxID string) (string, error) {
+
+	var files []string
+	err := filepath.Walk(kataGuestSharedDir, traverseFiles(&files))
+	if err != nil {
+		return "", err
+	}
+	for _, file := range files {
+		logrus.Printf("Files are %v", file)
+		if strings.Contains(file, sandboxID) && strings.Contains(file, "resolv.conf") {
+			return file, nil
+		}
+	}
+
+	return "", nil
+}
+
+func Mounted(mountpoint string) bool {
+	res, _ := exec.Command("sh", "-c", "mount | grep /etc/resolv.conf").Output()
```

Comment:
please don't use `sh`, `mount` and `grep`, instead read the file and parse it

Reply:
Yes please try to use a system call instead of executing a binary.